### PR TITLE
fix(core): support eip-1559 and eip-155 in wrw

### PR DESCRIPTION
**Pull request by sachushajiabraham**
add eip-1559 and replay protection options for unsigned json
which enables to properly construct replay protected eth tx from
ovc

Ticket: BG-44045

### CODE REVIEW

**Review by barathcj — DISMISSED** [review]

File: modules/core/src/v2/coins/eth.ts [hunk 1]
```diff
@@ -207,11 +219,8 @@ export interface RecoverOptions {
   krsProvider?: string;
   gasPrice?: number;
   gasLimit?: number;
-  eip1559?: { maxPriorityFeePerGas: number; maxFeePerGas: number };
-  replayProtectionOptions?: {
-    chain: string | number;
-    hardfork: string;
-  };
+  eip1559?: EIP1599;
+  replayProtectionOptions?: ReplayProtectionOptions;
```
Comment (mmcshinsky-bitgo) [inline modules/core/src/v2/coins/eth.ts:223]: Can we update BuildTransactionParams on line 226 to reflect and standardize the use of the newly added interfaces?

Comment (sachushajiabraham) [reply modules/core/src/v2/coins/eth.ts:221]: Good catch @mmcshinsky-bitgo , Updated it.

File: modules/core/src/v2/coins/eth.ts [hunk 2]
```diff
@@ -123,13 +123,20 @@ interface Recipient {
   data?: string;
 }
 
+interface EIP1599 {
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/eth.ts:126]: EIP number is 1559, not 1599

**Review by tylerlevine — CHANGES_REQUESTED** [review]
just one minor fix, otherwise lgtm

File: modules/core/src/v2/coins/eth.ts [hunk 3]
```diff
@@ -822,16 +826,19 @@ export class Eth extends BaseCoin {
...
+   * @param eip1559
+   * @param replayProtectionOptions
+   * @returns {Promise<OfflineVaultTxInfo>}
    */
   async formatForOfflineVault(
     txInfo: UnformattedTxInfo,
     ethTx: EthTxLib.Transaction | EthTxLib.FeeMarketEIP1559Transaction,
     userKey: string,
     backupKey: string,
     gasPrice: Buffer,
```
[the remark this answers is not in the document]
Comment (sachushajiabraham) [reply modules/core/src/v2/coins/eth.ts:838]: Left it as is since I am not sure whether any eth like chains which we support is not implementing eip 1559

**Review by wesgraham — APPROVED** [review]

**Pull request merged**